packages/domain/lib/usecase/usecase.dart

```diff
@@ -1,7 +1,29 @@
+import 'package:domain/validator/validator.dart';
+import 'package:domain/enum/validation_error_type.dart';
+
 abstract class UseCase<Output> {
   Output call();
 }
 
 abstract class UseCaseParams<Params, Output> {
   Output call(Params params);
 }
+
+abstract class BaseValidationUseCase<Params, Output>
+    implements UseCaseParams<Params, Output> {
+  @override
+  Output call(Params params);
+
+  ValidationErrorType? getEnumByValidator(Validator? v) {
+    if (v is RequiredField) {
+      return ValidationErrorType.requiredTypeError;
+    }
+    if (v is MinLength) {
+      return ValidationErrorType.minLengthTypeError;
+    }
+    if (v is RegEx) {
+      return ValidationErrorType.regexTypeError;
+    }
+    return null;
+  }
```

Comment:
move to enum ext

Reply:
it should be Validator extension then, bc I return enum ValidationErrorType based on Validator type

Reply:
@sergei-mikhailovskii-idf is this ok approach with baseValidationUseCase?

Reply:
up to u. as for me it's ok, since all validation usecases will contain such logic. but variant with extension/mapper is also ok

Reply:
what's the problem?